src/lua/fiber.lua

```diff
@@ -98,28 +98,23 @@ local function worker_f()
             stall()
         end
         worker_next_task = task.next
-        task.f(task.arg)
+        pcall(task.f, task.arg)
         fiber.sleep(0)
     end
 end
 
 local worker_name = 'tasks_worker_fiber'
 
-local function worker_safe_f()
-    pcall(worker_f)
-    -- Worker_f never returns. If the execution is here, this
-    -- fiber is probably canceled and now is not able to sleep.
-    -- Create a new one.
-    worker_fiber = fiber.new(worker_safe_f)
-    fiber_set_system(worker_fiber, true)
-    worker_fiber:name(worker_name)
-end
-
-worker_fiber = fiber.new(worker_safe_f)
-fiber_set_system(worker_fiber)
+worker_fiber = fiber.new(worker_f)
+fiber_set_system(worker_fiber, true)
+worker_fiber:set_joinable(true)
 worker_fiber:name(worker_name)
 
 local function worker_schedule_task(f, arg)
+    if worker_fiber == nil then
+        pcall(f, arg)
+        return
+    end
```

Comment:
Is it really necessary?

Reply:
Lua is still working after worker fiber shutdown so we may receive GC tasks.

Reply:
Discussed f2f and decided to drop this hunk and don't pursue executing all the tasks on Tarantool shutdown. We changed swim to work without this low priority worker. And in case of coio we may only leak a file descriptor.